subsys/bluetooth/audio/csis_client.c

```diff
@@ -140,6 +143,90 @@ static struct bt_csis *lookup_instance_by_set_info(const struct bt_csis_client_s
 	return NULL;
 }
 
+static struct bt_csis *get_next_active_instance(void)
+{
+	struct bt_csis_client_set_member *member;
+	struct bt_csis *inst;
+
+	member = active.members[active.members_handled];
+
+	inst =  lookup_instance_by_set_info(member, active.info);
+	if (inst == NULL) {
+		BT_DBG("Failed to lookup instance by set_info %p", active.info);
+	}
+
+	return inst;
+}
+
+static int member_rank_compare_asc(const void *m1, const void *m2)
+{
+	const struct bt_csis_client_set_member *member_1 =
+		*(const struct bt_csis_client_set_member **)m1;
+	const struct bt_csis_client_set_member *member_2 =
+		*(const struct bt_csis_client_set_member **)m2;
+	struct bt_csis *inst_1;
+	struct bt_csis *inst_2;
+
+	inst_1 = lookup_instance_by_set_info(member_1, active.info);
+	inst_2 = lookup_instance_by_set_info(member_2, active.info);
+
+	if (inst_1 == NULL) {
+		BT_ERR("inst_1 was NULL for member %p", member_1);
+		return 0;
+	}
+
+	if (inst_2 == NULL) {
+		BT_ERR("inst_2 was NULL for member %p", member_2);
+		return 0;
+	}
+
+	return inst_1->cli.rank - inst_2->cli.rank;
+}
+
+static int member_rank_compare_desc(const void *m1, const void *m2)
+{
+	/* If we call the "compare ascinding" function with the members
+	 * reversed, it will work as a descending comparison
+	 */
+	return member_rank_compare_asc(m2, m1);
+}
+
+static void active_members_store_ordered(struct bt_csis_client_set_member *members[],
```

Comment:
Looking through the PR, it doesn't seem like the original array of pointers is modified.  Is there a reason for this not being const?

Reply:
Essentially it is because `bt_csis_client_set_member` contains the `bt_conn`, and all the lower layer APIs (including GATT) does not have the `bt_conn` parameter as const. 

Besides that, we also have that `bt_csis_client_set_member.insts[x].csis` that also need to be non-const as that will also change during the procedures. 

Some of this API is changed again in https://github.com/zephyrproject-rtos/zephyr/pull/45164 to accommodate the CAP initiator, so some of this will unfortunately be changed again.

Reply:
ok, thanks - LGTM then